cmd/main.go

```diff
@@ -99,9 +103,24 @@ func main() {
 		if err := iptables.AddRule(s.AppPort, s.MetadataAddress, s.HostInterface, s.HostIP); err != nil {
 			log.Fatalf("%s", err)
 		}
+	} else if s.RemoveIPTablesRuleOnExit {
+		log.Fatalf("You cannot use --remove-iptables-on-exit without also specifying --iptables")
 	}
 
-	if err := s.Run(s.APIServer, s.APIToken, s.Insecure); err != nil {
-		log.Fatalf("%s", err)
+	signalChan := make(chan os.Signal)
+	go func() {
+		if err := s.Run(s.APIServer, s.APIToken, s.Insecure); err != nil {
+			log.Errorf("%s", err)
+			signalChan <- syscall.SIGABRT // On error, just quit now by faking a signal
```

Comment:
`SIGABRT` is not passed to `signal.Notify` below, is that going to work properly?

Reply:
`SIGABRT` just fakes an o/s signal so we force an exit in the event the goroutine was unable to start the server. (i.e. so `<-signalChan` a little further down won't block)  Kube can then get on with its job and restart this...

There are certainly better ways of handling this, but would be a somewhat more invasive PR - happy to revisit if needed...